packages/react/src/components/Checkbox/Checkbox.js

```diff
@@ -45,7 +45,7 @@ const Checkbox = React.forwardRef(function Checkbox(
           onChange(evt.target.checked, id, evt);
         }}
         className={`${prefix}--checkbox`}
-        id={id}
+        aria-labelledby={id}
```

Comment:
Any reason for this change? Would have thought that the label would still refer to this node even if the DOM order is flipped

Reply:
it was something to do with how it read. honestly, I can't recall and I can just change it back 😆 . There has been some good updates to checkbox while this PR was in flight anyway and I wouldnt want this change to make some newer change null or worse